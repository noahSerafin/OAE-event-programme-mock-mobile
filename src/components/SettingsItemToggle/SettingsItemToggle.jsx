import React from "react";
import "./SettingsItemToggle.scss";

const SettingsItemToggle = (props) => {
  const { settingsLabel, onChange, checked } = props;

  return (
    <div className="settings-item-toggle">
      <span className="settings-item-toggle__label">{settingsLabel}</span>
      <label className="settings-item-toggle__switch">
        <input
          className="settings-item-toggle__checkbox"
          type="checkbox"
          onChange={onChange}
          checked={checked}
        />
        <span className="settings-item-toggle__slider"></span>
      </label>
    </div>
  );
};

export default SettingsItemToggle;
