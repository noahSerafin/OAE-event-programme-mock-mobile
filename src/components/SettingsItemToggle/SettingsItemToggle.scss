@use "../../assets/sass/variables" as *;

.settings-item-toggle {
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid var(--hr-color);
  margin: 0 2.5rem;
  padding: 1rem 0;
  position: relative;
  top: 1em;
  &__label {
    font-weight: 700;
    margin: 0;
  }
  &__switch {
    width: 2.813rem;
    height: 1.5rem;
    position: relative;
    display: inline-block;
  }
  &__checkbox {
    opacity: 0;
    width: 0;
    height: 0;
    &:checked + .settings-item-toggle__slider {
      background-color: var(--theme-color-primary);
    }
    &:checked + .settings-item-toggle__slider:before {
      transform: translateX(1.2rem);
    }
  }
  &__slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: #eaeaea;
    transition: 0.4s;
    border-radius: 34px;
    &:before {
      position: absolute;
      content: "";
      height: 1.3rem;
      width: 1.3rem;
      left: 2px;
      bottom: 2px;
      background-color: white;
      transition: 0.4s;
      border-radius: 50%;
    }
  }
}
