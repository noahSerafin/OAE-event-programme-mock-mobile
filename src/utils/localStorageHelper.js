export const getDarkTheme = () => {
  const isDarkMode = localStorage.getItem("darkMode");
  return isDarkMode === "true";
};
export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};

export const getFontSize = () => {
  const fontSize = localStorage.getItem("fontSize");
  return fontSize;
};

export const setFontSize = (fontSize) => {
  console.log(fontSize);
  localStorage.setItem("fontSize", fontSize);
};
