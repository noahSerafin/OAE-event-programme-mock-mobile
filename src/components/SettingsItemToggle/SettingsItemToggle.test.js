import { customRender } from "../../utils/testUtils";
import SettingsItemToggle from "./SettingsItemToggle";

test("renders settingsItemToggle component", () => {
  const { container } = customRender(<SettingsItemToggle />);
  expect(container).toMatchSnapshot();
});
