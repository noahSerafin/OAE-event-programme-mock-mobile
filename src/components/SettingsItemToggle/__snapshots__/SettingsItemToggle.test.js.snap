// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders settingsItemToggle component 1`] = `
<div>
  <div
    class="settings-item-toggle"
  >
    <span
      class="settings-item-toggle__label"
    />
    <label
      class="settings-item-toggle__switch"
    >
      <input
        class="settings-item-toggle__checkbox"
        type="checkbox"
      />
      <span
        class="settings-item-toggle__slider"
      />
    </label>
  </div>
</div>
`;
